drivers/resolver/ad2s1210/ad2s1210.c

```diff
@@ -0,0 +1,408 @@
+*
+* @param dev - Device descriptor.
+* @param addr - Address of the register to write
+* @param val - value to write
+*
+* @return Returns negative error code or 0 in case of success.
+*******************************************************************************/
+int32_t ad2s1210_reg_write(struct ad2s1210_dev *dev, uint8_t addr, uint8_t val)
+{
+	uint8_t buf;
+	int ret;
+
+	if (addr < AD2S1210_REG_MIN)
+		return -EINVAL;
+
+	ret = ad2s1210_set_mode_pins(dev, MODE_CONFIG);
+	if (ret)
+		return ret;
+
+	buf = addr;
+	ret = no_os_spi_write_and_read(dev->spi_desc, &buf, 1);
+	if (ret)
+		return ret;
+
+	buf = val;
+	return no_os_spi_write_and_read(dev->spi_desc, &buf, 1);
+}
+
+/*******************************************************************************
+* @brief Read a device register
+*
+* @param dev - Device descriptor.
+* @param addr - Address of the register to read
+* @param val - pointer to store register value
+*
+* @return Returns negative error code or 0 in case of success.
+*******************************************************************************/
+int32_t  ad2s1210_reg_read(struct ad2s1210_dev *dev, uint8_t addr, uint8_t *val)
+{
+	uint8_t buf;
+	int32_t ret;
+
+	if (addr < AD2S1210_REG_MIN)
+		return -EINVAL;
+
+	ret = ad2s1210_set_mode_pins(dev, MODE_CONFIG);
+	if (ret)
+		return ret;
+
+	buf = addr;
+	ret = no_os_spi_write_and_read(dev->spi_desc, &buf, 1);
+	if (ret)
+		return ret;
+
+	buf = addr;
+	ret = no_os_spi_write_and_read(dev->spi_desc, &buf, 1);
+	if (ret)
+		return ret;
+
+	*val = buf;
+	return 0;
+}
+
+/*******************************************************************************
+* @brief Read a device register
+*
+* @param dev - Device descriptor.
+* @param mode - enum that contains the mode to set the device
+*
+* @return Returns negative error code or 0 in case of success.
+*******************************************************************************/
+static int32_t ad2s1210_set_mode_pins(struct ad2s1210_dev *dev,
+				      enum ad2s1210_mode mode)
+{
+	int32_t ret;
+
+	if (dev->mode == mode)
+		return 0;
+
+	ret = no_os_gpio_set_value(dev->gpio_a0, mode & NO_OS_BIT(0));
+	if (ret)
+		return ret;
+
+	ret = no_os_gpio_set_value(dev->gpio_a1, mode & NO_OS_BIT(1));
+	if (ret)
+		return ret;
+
+	dev->mode = mode;
+	return 0;
+}
+
+/*******************************************************************************
+* @brief Set device bit resolution 10, 12, 14, or 16 bits
+*
+* @param dev - Device descriptor.
+* @param resolution - resolution value
+*
+* @return Returns negative error code or 0 in case of success.
+*******************************************************************************/
+static int32_t ad2s1210_set_resolution(struct ad2s1210_dev *dev,
+				       uint8_t resolution)
+{
+	int32_t ret;
+	uint8_t control;
+
+	ret = ad2s1210_reg_read(dev, AD2S1210_REG_CONTROL, &control);
+	if (ret)
+		return ret;
+
+	control &= ~(AD2S1210_REG_CONTROL_RES_MASK);
+	switch (resolution) {
+	case 10:
```

Comment:
what does 10, 12, 14, 16 mean? add suggestive enum or macros.

Reply:
is the actual resolution 10,12,14,16 bits. ill add a macro if it makes it more clear.